app/portainer/settings/edge-compute/FDOProfilesDatatable/FDOProfilesDatatable.tsx

```diff
@@ -0,0 +1,155 @@
+import { useTable, usePagination, useSortBy } from 'react-table';
+import { useRowSelectColumn } from '@lineup-lite/hooks';
+import { FDOProfilesDatatableActions } from 'Portainer/settings/edge-compute/FDOProfilesDatatable/FDOProfilesDatatableActions';
+import { SelectedRowsCount } from 'Portainer/components/datatables/components/SelectedRowsCount';
+import { PaginationControls } from 'Portainer/components/pagination-controls';
+import { TableFooter } from 'Portainer/components/datatables/components/TableFooter';
+import { useTableSettings } from 'Portainer/components/datatables/components/useTableSettings';
+import { useRowSelect } from 'Portainer/components/datatables/components/useRowSelect';
+
+import { Profile } from '@/portainer/hostmanagement/fdo/model';
+import PortainerError from '@/portainer/error';
+import {
+  Table,
+  TableActions,
+  TableContainer,
+  TableHeaderRow,
+  TableRow,
+  TableTitle,
+} from '@/portainer/components/datatables/components';
+import { FDOProfilesTableSettings } from '@/edge/devices/types';
+import { useFDOProfiles } from '@/portainer/settings/edge-compute/FDOProfilesDatatable/useFDOProfiles';
+
+import { useColumns } from './columns';
+
+export function FDOProfilesDatatable() {
+  const { settings, setTableSettings } =
+    useTableSettings<FDOProfilesTableSettings>();
+  const columns = useColumns();
+
+  const { isLoading, profiles, error } = useFDOProfiles();
+
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    page,
+    prepareRow,
+    selectedFlatRows,
+    gotoPage,
+    setPageSize,
+    state: { pageIndex, pageSize },
+  } = useTable<Profile>(
+    {
+      defaultCanFilter: false,
+      columns,
+      data: profiles,
+      initialState: {
+        pageSize: settings.pageSize || 10,
+        sortBy: [settings.sortBy],
+      },
+      isRowSelectable() {
+        return true;
+      },
+      selectColumnWidth: 5,
+    },
+    useSortBy,
+    usePagination,
+    useRowSelect,
+    useRowSelectColumn
+  );
+
+  const tableProps = getTableProps();
+  const tbodyProps = getTableBodyProps();
```

Comment:
this whole code is copy pasted, something to think about...

Reply:
mostly it its, but there are some small differences, are you thinking this could be moved to a "helper" hook? something like useTableBase ?

Reply:
yes, I don't know :) just thinking out loud.